Финальный проект 5 спринта

AlexeyLandik opened the pull request.

AlexeyLandik committed:
Финальный проект 5 спринта

irisqul commented on README.md:
Можно лучше: будет здорово описать в README инструкцию по запуску проекта

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,13 @@
+    options = Options()
+    options.add_argument('--window-size=1920,1080')
+    driver = webdriver.Chrome(options=options)
+    driver.get("https://stellarburgers.nomoreparties.site/")
```
Можно лучше: вынести урлы в файл с константами и переиспользовать

irisqul commented on conftest.py:
```diff
@@ -0,0 +1,13 @@
+
+
+@pytest.fixture(scope='function')
+def driver():
```
Отлично: написана фикстура

irisqul commented on locators.py:
```diff
@@ -0,0 +1,26 @@
+from selenium.webdriver.common.by import By
+
+
```
Отлично: локаторы вынесены в отдельный файл

irisqul commented on tests/test_constructor.py:
```diff
@@ -0,0 +1,26 @@
+        driver.find_element(*TestLocators.CHAPTER_BRAD).click()
+        current_chapter = driver.find_element(*TestLocators.CURRENT_CHAPTER).text
+
+        assert current_chapter == 'Булки'
```
Нужно исправить: наличие текста у этих элементов не говорит о том, что переход на таб произошел. У элемента при клике добавляется новый класс. Следует проверять его

irisqul commented on tests/test_personal_account.py:
```diff
@@ -0,0 +1,49 @@
+        WebDriverWait(driver, 3).until(
+            expected_conditions.visibility_of_element_located(TestLocators.BUTTON_EXIT_IN_PROFILE))
+        driver.find_element(*TestLocators.BUTTON_EXIT_IN_PROFILE).click()
+        login_page = 'https://stellarburgers.nomoreparties.site/login'
```
up, лучше все урлы вынести в константы, тогда код будет более поддерживаемым

AlexeyLandik committed:
1. README.md - добавлена инструкция для запуска
2. config.py - создан класс с константами
3. test_constructor.py - исправлены тесты

AlexeyLandik commented:
Добавлен новый коммит с правками:
1. README.md - добавлена инструкция для запуска
2. config.py - создан класс с константами
3. test_constructor.py - исправлены тесты
